app/backend/approaches/chatreadretrieveread.py

```diff
@@ -150,17 +146,17 @@ async def run_until_final_call(
             },
         )
 
-        extra_info = {
-            "data_points": {"text": text_sources},
-            "thoughts": [
-                ThoughtStep(
-                    "Prompt to generate search query",
-                    query_messages,
-                    (
-                        {"model": self.chatgpt_model, "deployment": self.chatgpt_deployment}
-                        if self.chatgpt_deployment
-                        else {"model": self.chatgpt_model}
-                    ),
+        extra_info = ExtraInfo(
+            DataPoints(text=text_sources),
+            thoughts=[
+                self.create_generate_thought_step(
```

Comment:
Oh I think now I get the function name, as that function only generates thought steps for chat completions.
Perhaps format_thought_step_for_chat_completion would be clear?